beluga/include/beluga/sensor/data/regular_grid.hpp

```diff
@@ -74,8 +74,12 @@ class BaseRegularGrid2 : public ciabatta::ciabatta_top<Derived> {
    * \return Grid cell coordinates.
    */
   [[nodiscard]] Eigen::Vector2i cell_near(double x, double y) const {
-    const auto xi = static_cast<int>(std::floor(x / this->self().resolution()));
-    const auto yi = static_cast<int>(std::floor(y / this->self().resolution()));
+    const auto inv_resolution = 1 / this->self().resolution();
+    const auto scaled_x = x * inv_resolution;
+    const auto scaled_y = y * inv_resolution;
+    // don't use floor() due to performance reasons
+    const auto xi = static_cast<int>(scaled_x < 0.0 ? -2.0 : scaled_x);
+    const auto yi = static_cast<int>(scaled_y < 0.0 ? -2.0 : scaled_y);
```

Comment:
Hmm, I don't follow why there is a `-2.0` in there.

Reply:
No particular reason. Any negative that does not cast into 0 will do, because we don't care about negatives since they are not in the grid. This would probably work as well:
```
const auto xi = static_cast<int>(scaled_x < 0.0 ? scaled_x - 1.0 : scaled_x);
```
but [it's not exactly the same](https://stackoverflow.com/a/825916) as `std::floor` and wastes cycles on negatives.

Reply:
Makes sense, can we document the return value in the doxygen block of this method? And adding a local comment on why we use `-2.0` wouldn't hurt either.

Reply:
Indeed. I'll add it.

Reply:
Thinking it over, 
```
const auto xi = scaled_x < 0.0 ?  -1 : static_cast<int>(scaled_x);
```
looks far less esoteric and should be just as performant. Still requires comments though.